astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,469 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Combine 3 images to produce a properly-scaled RGB image following Lupton et al. (2004).
+
+The three images must be aligned and have the same pixel scale and size.
+
+For details, see : http://adsabs.harvard.edu/abs/2004PASP..116..133L
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import numpy as np
+from . import ZScaleInterval
+
+
+__all__ = ['make_lupton_rgb']
+
+
+try:
+    import scipy.misc
+    scipy.misc.imresize  # checking if it exists
+    HAVE_SCIPY_MISC = True
+except (ImportError, AttributeError):
+    HAVE_SCIPY_MISC = False
+
+# NOTE: these methods would have come from LSST C++ code. They won't be available
+# in astropy until they are converted somehow.
+# from lsst.afw.display.displayLib import replaceSaturatedPixels, getZScale
+
+
+def compute_intensity(image_r, image_g=None, image_b=None):
+    """
+    Return a naive total intensity from the red, blue, and green intensities.
+
+    Parameters
+    ----------
+    image_r : `~numpy.ndarray`
+        Intensity of image to be mapped to red; or total intensity if ``image_g``
+        and ``image_b`` are None.
+    image_g : `~numpy.ndarray`, optional
+        Intensity of image to be mapped to green.
+    image_b : `~numpy.ndarray`, optional
+        Intensity of image to be mapped to blue.
+
+    Returns
+    -------
+    intensity : `~numpy.ndarray`
+        Total intensity from the red, blue and green intensities, or ``image_r``
+        if green and blue images are not provided.
+    """
+    if image_g is None or image_b is None:
+        if not (image_g is None and image_b is None):
+            raise ValueError("please specify either a single image "
+                             "or red, green, and blue images.")
+        return image_r
+
+    intensity = (image_r + image_g + image_b)/3.0
+
+    # Repack into whatever type was passed to us
+    return np.array(intensity, dtype=image_r.dtype)
+
+
+class Mapping(object):
+    """Baseclass to map red, blue, green intensities into uint8 values."""
+
+    def __init__(self, minimum=None, image=None):
+        """
+        Create a mapping.
+
+        Parameters
+        ----------
+        minimum : float or sequence(3)
+            Intensity that should be mapped to black (a scalar or array for R, G, B).
+        image : `~numpy.ndarray`, optional
+            The image to be used to calculate the mapping.
+            If provided, it is also used as the default for make_rgb_image().
+        """
+        self._uint8Max = float(np.iinfo(np.uint8).max)
+
+        try:
+            len(minimum)
+        except TypeError:
+            minimum = 3*[minimum]
+        if len(minimum) != 3:
+            raise ValueError("please provide 1 or 3 values for minimum.")
+
+        self.minimum = minimum
+        self._image = np.asarray(image)
```

Comment:
This `_image` attribute seems useless, all images should be passed to `make_rgb_image`, so there is no need to have 2 ways to do this.
*edit:* after looking at the Mapping classes, it seems useful for `AsinhZScaleMapping`.